komorebi-bar/src/widgets/applications.rs

```diff
@@ -0,0 +1,326 @@
+        std::thread::spawn(move || {
+            if let Err(e) = Command::new("cmd").args(["/C", &command_string]).spawn() {
+                tracing::error!("Failed to launch command '{}': {}", command_string, e);
+            }
+        });
+    }
+}
+
+/// Holds decoded image data to be used as an icon in the UI.
+#[derive(Clone, Debug)]
+pub enum Icon {
+    /// RGBA image used for rendering the icon.
+    Image(RgbaImage),
+    /// Text-based icon, e.g. from a font like Nerd Fonts.
+    Text(String),
+}
+
+impl Icon {
+    /// Attempts to create an `Icon` from the given `AppConfig`.
+    /// Loads the image from a specified icon path or extracts it from the application's
+    /// executable if the command points to a valid executable file.
+    #[inline]
+    pub fn try_from(config: &AppConfig) -> Option<Self> {
+        if let Some(icon) = config.icon.as_deref().map(str::trim) {
+            if !icon.is_empty() {
+                let path = std::path::Path::new(icon);
+                if path.is_file() {
+                    match image::open(path).as_ref().map(DynamicImage::to_rgba8) {
+                        Ok(image) => return Some(Icon::Image(image)),
+                        Err(err) => {
+                            tracing::error!("Failed to load icon from {}, error: {}", icon, err)
+                        }
+                    }
+                } else {
+                    return Some(Icon::Text(icon.to_owned()));
+                }
+            }
+        }
+
+        let path = std::path::Path::new(&config.command);
+        if path.is_file() {
+            return windows_icons::get_icon_by_path(&config.command).map(Icon::Image);
+        }
+        None
+    }
+
+    /// Renders the icon in the given `Ui` context with the specified size.
+    #[inline]
+    pub fn draw(&self, ctx: &Context, ui: &mut Ui, icon_config: &IconConfig) {
+        match self {
+            Icon::Image(image) => {
+                Frame::NONE
+                    .inner_margin(Margin::same(ui.style().spacing.button_padding.y as i8))
+                    .show(ui, |ui| {
+                        ui.add(
+                            Image::from(&img_to_texture(ctx, image))
+                                .maintain_aspect_ratio(true)
+                                .fit_to_exact_size(Vec2::splat(icon_config.size)),
+                        );
+                    });
+            }
+            Icon::Text(icon) => {
+                let rich_text = RichText::new(icon)
+                    .font(icon_config.font_id.clone())
+                    .size(icon_config.size)
+                    .color(icon_config.color);
+                ui.add(Label::new(rich_text).selectable(false));
+            }
+        }
+    }
+
+    /// Draws a fallback icon when the specified icon cannot be loaded.
+    /// Displays a simple crossed-out rectangle as a placeholder.
+    #[inline]
+    pub fn draw_fallback(ui: &mut Ui, icon_size: Vec2) {
```

Comment:
This is not quite the same as the `windows_icons_fallback`

`windows_icons_fallback` reverts to an older variant of the icon lookup algorithm which often succeeds in cases when the `windows_icons` algorithm fails

What you have here seems more like a fallback to that 2nd-level fallback

Reply:
Fixed